include/meta/meta.hpp

```diff
@@ -934,13 +942,31 @@ namespace meta
     template <typename... Args>
     using transform = eval<detail::transform_<Args...>>;
 
+    ////////////////////////////////////////////////////////////////////////////
+    // filter
+    /// \cond
+    template <typename Predicate> struct filter_
+    {
+        template <typename State, typename A>
+        using apply = meta::if_<meta::apply<Predicate, A>,
+                                meta::push_back<State, A>, State>;
+    };
+    /// \endcond
+
+    /// Returns a new meta::list where all the elements of \p List A that
+    /// satisfy the Metafunction Class \p Predicate such that
+    /// `apply<Pred,A>::%value` is \c true have been removed.
+    template <typename List, typename Predicate>
+    using filter = meta::foldl<List, meta::list<>, filter_<Predicate>>;
```

Comment:
@pfultz2 thanks I'll take a look at it in the upcoming days. We'll see where that leads. 

If you want to propose such a feature for meta feel free to send a pull-request and we can move the discussion over there.